### ISSUE TITLE
explicitADmode is overwritten by silent configuration update

### ISSUE THREAD

**Issue by astehlik** [helhum/typo3-config-handling#26]
This is what seems to happen:

The migration for the `explicitADmode` Setting (see https://forge.typo3.org/issues/94721) will check, if there is a value configured in the `LocalConfiguration.php` for `BE/explicitADmode`. If this is not the case, the value is set to `explicitDeny` (the old default value).

The value will always be empty (even if set to `explicitADmode`) in the config, because the `ConfigCleaner` throws out all values that are identical to those in the `DefaultConfiguration.php` of the TYPO3 core.

The only solution I can see is that the `ConfigCleaner` does not remove the `BE/explicitADmode` config path, even if it is identical.

I'm happy to provide a PR, if this is a valid solution for you :)

Cheers
Alex

**Comment by helhum** [comment]
> This is what seems to happen:

Yep. This is an issue present for a long time, but so far luckily did not lead to any bug.

> The only solution I can see is that the `ConfigCleaner` does not remove the `BE/explicitADmode` config path, even if it is identical.

Hm, na. The issue isn't this specific config value or the change in TYPO3, but that there (until now), there is no API to just return the configuration provided by the user. With this bug now in existence, it is time to finally add this API.

**Comment by helhum** [comment]
@astehlik can you check, whether the PR solves your issue?
